### PR TITLE
chore(legacy-plugin-chart-map-box): bump supercluster to v8

### PR DESCRIPTION
**Pull request by birkskyum**
### SUMMARY
Part of:
- #30307 

Modernizing the legacy-plugin-chart-map-box reduces the diff towards MapLibre.

This PR updates Supercluster from v4 to v8, which [improves performance and memory use](https://github.com/mapbox/supercluster/pull/223) for of this plugin.

### TESTING INSTRUCTIONS
- set the mapbox key in .env
- docker compose up
- open a mapbox chart, and it should still show clusters like below image

Screenshot 2024-09-17 at 11 15 56

@rusackas , how does the review process generally work for these plugins? Are there some specific people who cover this part of the code?

### CODE REVIEW

**Review by mistercrunch — APPROVED** [review]

**Comment by rusackas** [comment]
@birkskyum to test/review these PRs, we typically need to check the plugin. There are a few ways to do this:
1) starting an ephemeral env here so anyone can test it (needs to be done by a committer)
2) pulling the change locally and running Superset (better in this case so we can add a mapbox key)
3) pulling the change locally and running Storybook

I'll start with #1 here before I approve. @mistercrunch was your review a code review, or did you test/validate the upgrade?

**Comment by mistercrunch** [comment]
I only reviewed the 3 lines, seemed benign, didn't test locally

**Comment by rusackas** [comment]
Tested, works great! Thanks, @birkskyum!

**Comment by rusackas** [comment]
No idea why ephemeral environments aren't working again. Any ideas, @mistercrunch? Did those move to supersetbot or something?

**Pull request merged**
